src/connection/xmpp/Connector.ts

```diff
@@ -40,7 +42,7 @@ export default class Connector {
       } else if (connectionArgs.length === 3 || connectionArgs.length === 4) {
          this.connectionArgs = connectionArgs;
 
-         let type = /^wss?:/.test(connectionArgs[1]) ? TYPE.WEBSOCKET : TYPE.BOSH;
+         let type = this.checkForWebsocketUrl(connectionArgs) ? TYPE.WEBSOCKET : TYPE.BOSH;
```

Comment:
Why are you testing all arguments?

Reply:
i saw that sometimes the url was on another index.. dont now why... maybe it was a caching bug...
will take a look again soon

Reply:
ok now allways checking parameter 0